src/Bicep.Core/Parser/Parser.cs

```diff
@@ -410,6 +449,33 @@ private IdentifierSyntax Identifier(DiagnosticBuilder.ErrorBuilderDelegate error
             return new IdentifierSyntax(identifier);
         }
 
+        private IdentifierSyntaxBase IdentifierWithRecovery(DiagnosticBuilder.ErrorBuilderDelegate errorFunc, params TokenType[] terminatingTypes)
+        {
+            var identifierOrSkipped = this.WithRecovery(
+                () => Identifier(errorFunc),
+                RecoveryFlags.None,
+                terminatingTypes);
+
+            switch (identifierOrSkipped)
+            {
+                case IdentifierSyntax identifier:
+                    return identifier;
+
+                case SkippedTriviaSyntax skipped when skipped.Elements.Any() == false:
+                    var position = this.reader.Peek().Span.Position;
+                    return new MalformedIdentifierSyntax(new[] {CreateMissingToken(position)}, skipped.Diagnostics);
```

Comment:
I don't doubt you have a good reason for creating `MalformedIdentifierSyntax` rather than using `SkippedTriviaSyntax`, but what is that reason?

Reply:
Yeah the problem is that we use `IdentifierName` property of IdentifierSyntax all over the place, so `SkippedTriviaSyntax` will require adding extra checks all over the place. Having a class with the property is much more cleaner. However, I'm coming around to just modify `IdentifierSyntax` to capture a list of tokens instead of just one and have it return the right identifier name based on some conditions.

Reply:
It's gone now.